entry/streamer-entry

```diff
@@ -0,0 +1,208 @@
+
+function main() {
+  local uid=0
+  local w=${K8S_ENV_DISPLAY_RESOLUTION_X:-1280}
+  local h=${K8S_ENV_DISPLAY_RESOLUTION_Y:-720}
+  local y=${ICR_ON:-n}
+  local user_count=${K8S_ENV_USER_COUNT_PER_ANDROID:-1}
+  local dev=${DEVICE:-/dev/dri/renderD128}
+  local cqp=${K8S_ENV_FORCE_CQP:-false}
+  local wdir="/opt/workdir"
+  local k8s=${K8S_ENV:-"false"}
+  local k8s_stateless=${K8S_ENV_STATELESS:-"false"}
+  local dev=${DEVICE:-/dev/dri/renderD128}
+  local id="$(id)"
+  local measure_latency="false"
+  local privileged_enable=${ENABLE_ENV_PRIVILEGED:-"false"}
+  local codec=${CODEC:-"avc"}
+  local icr_start_immediately=${ENV_ICR_START_IMMEDIATELY:-"false"}
+
+  while getopts ":i:u:lh" opt; do
+    case $opt in
+    i) id=$OPTARG ;;
+    u) uid=$OPTARG;;
+    l) measure_latency="true";;
+    h) help && exit 1 ;;
+    *) echo "no such option: $opt" && exit 1 ;;
+    esac
+  done
+  shift $((OPTIND - 1))
+
+  if [[ -n "$1" ]]; then
+    ip="$1"
+  else
+    err "signaling server ip is required"
+    help
+    exit 1
+  fi
+
+  dbg "id is $id"
+  dbg "ips is $ips"
+  dbg "port is $port"
+  dbg "measure_latency is $measure_latency"
+
+  local ips=$ip
+  local peerid=$[$user_count*$id+$uid]
+  local serverid=s$peerid
+  local clientid=c$peerid
+  local coturn_ip=${COTURN_IP:-"$ip"}
+  local ice_port_begin=${ICE_PORT_BEGIN:-"50000"}
+  local ice_port_end=${ICE_PORT_END:-"50999"}
+
+  local ipc files
+  if [[ "$k8s" == "true" ]]; then
+    ipc="/conn"
+    wdir="/conn"
+    nid=0
+    ips=${K8S_ENV_OWT_SERVER_IP:-owt}
+    port=${K8S_ENV_OWT_SERVER_PORT:-8095}
+    portOpt=" -p $port"
+    k8sOpt="--k8s 1"
+    local GPU_NODES=${GPU_NODES:-$(find /dev/dri -iname "renderD*" | wc -l)}
+    local INS_ON_NODE="$(expr $id % $GPU_NODES + 128)"
+    dev="/dev/dri/renderD$INS_ON_NODE"
+    if [[ "$k8s_stateless" == "true" ]]; then
+        wait_file /conn/stateless/id
+        dev="/dev/dri/renderD128"
+        IFS="," read -ra arr <<< "$(cat /conn/stateless/id)"
+        serverid=${arr[0]}
+        clientid=${arr[1]}
+        echo wait for icr container start up...
+        sleep 10
+    fi
+  else
+    ipc="/opt/workdir/ipc"
+    nid=$id
+  fi
+  if [[ $VIRTUAL_INPUT_NUM -gt 0 ]]; then
+    for ((i=0;i<$VIRTUAL_INPUT_NUM; i++)); do
+      if [[ "$k8s" == "true" ]]; then
+        files="input-pipe-$i"
+      elif [[ $user_count -gt 1 ]]; then
+        files="input-pipe${id}-$uid-$i"
+      else
+        files="input-pipe${id}-$i"
+      fi
+    done
+  fi
+  for f in $files; do
+    wait_file "${ipc}/${f}"
+  done
+
+  local opts="-s $ips  $portOpt $k8sOpt -d $dev --width $w --height $h -n $nid --server-peer-id $serverid --client-peer-id $clientid --workdir $wdir"
+
+  if [[ $user_count -gt 1 ]]; then
+      opts="$opts --enable-multi-user 1 --user $uid"
+  fi
+
+  if [ "$cqp" == "true" ]; then
+      opts="$opts --video-rc cqp"
+  fi
+
+  if [ "$y" == "y" ]; then
+      if [[ $user_count -gt 1 ]]; then
+          iIP=`cat "/opt/workdir/icr/icr"$id-$uid`
```

Comment:
who creates this file? Why this can't be done thru the command line option instead?